### DISCUSSION TITLE
Head elements I added in layout.js and _app.js don't appear in exported HTML files.

### DISCUSSION THREAD

**Discussion by Heein-Park** [vercel/next.js#13833] [Help]
![1](https://user-images.githubusercontent.com/25489873/83937026-260dde00-a804-11ea-9bf6-46e2eb88cdb7.png)
![2](https://user-images.githubusercontent.com/25489873/83937027-27d7a180-a804-11ea-977b-9a27631fb3b7.png)
![3](https://user-images.githubusercontent.com/25489873/83937029-27d7a180-a804-11ea-9c0c-b57c20d649dd.png)

Due to the memory shortage of my server, I adopt a static HTML generation strategy, in which I utilize npm run build & npm run export to create HTML results of my Next JS app and put those HTML files into my server.

However, head elements such as a title or meta don't appear in generated HTML files even if they behave normally in development mode (I assume that they won't be problematic in a regular Next JS app.)

Believe or not, when I put some of the head elements into _document.js HEAD, they appear. Nevertheless, migrating every head element into _document.js means I cannot assign open graph meta tags for individual pages and also I have to insert the title HTML element into _document.js, the act that should not be done in Next JS development.

I tried to search for any issue related to next/head along with Static HTML generation, but it seems to be minor discussion, therefore I post this to find out any solution.

**Comment by ivanbtrujillo** [comment]
I've the same issue than you. I've a blog and I'm trying to generate dynamic twitter cards based on each post data. What I've noticed is:

* when I use next/head in another component without create a _document, it works as expected BUT I cannot set the language of the page (I need a _documment for that).

* when I use next/head in another component AND a _document to set the languague of the page, the meta tags that are defined using next/head does not appear BUT If I add meta tags inside _document Head, they appear.

So I suspect that _document Head is overwriting next/head somehow. 

Any solution to avoid it?

I'm using **v9.4.4**. I've tried  **v9.4.5-canary.31** without success (same behavior)

Also I noticed that using next/head, If I run **build and start** or **export and executing an http-server**, the headers appears but after deploy to vercel, they don't. 🤔
